x-pack/plugins/uptime/server/rest_api/monitors/monitors_durations.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { UMServerLibs } from '../../lib/lib';
+import { UMRestApiRouteFactory } from '../types';
+
+export const createGetMonitorDurationRoute: UMRestApiRouteFactory = (libs: UMServerLibs) => ({
+  method: 'GET',
+  path: '/api/uptime/monitor/duration',
+  validate: {
+    query: schema.object({
+      monitorId: schema.string(),
+      dateStart: schema.maybe(schema.string()),
+      dateEnd: schema.maybe(schema.string()),
+    }),
+  },
+  options: {
```

Comment:
I'm thinking that the tags option should have `access:uptime` applied to it as an abstraction so it's not up to the implementer to specify, seems like an opportunity to fail. We can write a unit test to check for the presence of this tag as well, all our routes should have it. Not part of this PR, but it's highlighting lots of things we can tighten up!

Reply:
Tracking this in an issue: https://github.com/elastic/uptime/issues/154